ClientContext.cs

```diff
@@ -13,8 +10,6 @@ namespace AzureKeyVaultRecoverySamples
     /// </summary>
     public sealed class ClientContext
```

Comment:
None of this class is really used anymore. Perhaps just replace it with the `DefaultAzureCredential` directly in the client like we normally do in samples. I worry the unnecessary complexity here will just make people copy the design when simply passing a `DefaultAzureCredential` is so much simpler.